datasets/coached_conv_pref/coached_conv_pref.py

```diff
@@ -0,0 +1,175 @@
+
+_URLs = {"dataset": "https://storage.googleapis.com/dialog-data-corpus/CCPE-M-2019/data.json"}
+
+
+class CoachedConvPrefConfig(datasets.BuilderConfig):
+    """BuilderConfig for DialogRE"""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for DialogRE.
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(CoachedConvPrefConfig, self).__init__(**kwargs)
+
+
+class CoachedConvPref(datasets.GeneratorBasedBuilder):
+    """Coached Conversational Preference Elicitation Dataset to Understanding Movie Preferences"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        CoachedConvPrefConfig(
+            name="coached_conv_pref",
+            version=datasets.Version("1.1.0"),
+            description="Coached Conversational Preference Elicitation Dataset to Understanding Movie Preferences",
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "conversationId": datasets.Value("string"),
+                    "utterances": datasets.Sequence(
+                        {
+                            "index": datasets.Value("int32"),
+                            "speaker": datasets.features.ClassLabel(names=["USER", "ASSISTANT"]),
+                            "text": datasets.Value("string"),
+                            "segments": datasets.Sequence(
+                                {
+                                    "startIndex": datasets.Value("int32"),
+                                    "endIndex": datasets.Value("int32"),
+                                    "text": datasets.Value("string"),
+                                    "annotations": datasets.Sequence(
+                                        {
+                                            "annotationType": datasets.features.ClassLabel(
+                                                names=[
+                                                    "ENTITY_NAME",
+                                                    "ENTITY_PREFERENCE",
+                                                    "ENTITY_DESCRIPTION",
+                                                    "ENTITY_OTHER",
+                                                    "",
+                                                ]
+                                            ),
+                                            "entityType": datasets.features.ClassLabel(
+                                                names=[
+                                                    "MOVIE_GENRE_OR_CATEGORY",
+                                                    "MOVIE_OR_SERIES",
+                                                    "PERSON",
+                                                    "SOMETHING_ELSE",
+                                                    "",
```

Comment:
```suggestion
```
Let's keep only valid classes.
If for some reason one example has a missing label, you can use the `-1` value for this example

Reply:
@lhoestq I have made all the changes you proposed. Please take a look at it and let me know of any improvements.